### PR TITLE
[build][test] MXNet 1.7.0 Fix safety for pip

### PR DESCRIPTION
**Pull request by jeet4320**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: mxnet/inference/docker/1.7.0/py3/cu101/Dockerfile.gpu
Comment (arjkesh) [inline mxnet/inference/docker/1.7.0/py3/cu101/Dockerfile.gpu:96]: likely it's a good idea to restrict the major versions of both of these packages to avoid conflicts or breaking changes

File: mxnet/training/docker/1.7.0/py3/Dockerfile.cpu
Comment (arjkesh) [inline mxnet/training/docker/1.7.0/py3/Dockerfile.cpu:114]: Does the version of MXNet in this container no longer require a pip compatible with python2? I'm not sure we have tests for this. It's probably safe, but good to verify. pip 21.0 removed python2 support, latest version is 21.1.2

Comment (jeet4320) [reply mxnet/training/docker/1.7.0/py3/Dockerfile.cpu:114]: no, latest pip is fine,  I have confirmed internally

File: mxnet/inference/docker/1.7.0/py3/cu101/Dockerfile.gpu
Comment (jeet4320) [reply mxnet/inference/docker/1.7.0/py3/cu101/Dockerfile.gpu:96]: awscli > 2 has breaking changes, so <2 and boto3 is always latest we install. It is as per all other latest framework version dockerfiles as well.

File: mxnet/buildspec.yml
Comment (arjkesh) [inline mxnet/buildspec.yml:42]: Are these new image baseline sizes coming from the mx 1.6 build spec? 

We should likely follow the convention we have in TF and version these buildspecs. Would you mind creating a copy of this as 1-7, and an additional file for the 1-8 MXNet so people have easy access to what the build spec should be?

Comment (jeet4320) [reply mxnet/buildspec.yml:42]: they are temp changes to run 1.7 on PR

File: mxnet/inference/docker/1.7.0/py3/cu101/Dockerfile.gpu
Comment (arjkesh) [reply mxnet/inference/docker/1.7.0/py3/cu101/Dockerfile.gpu:96]: Given the current state of boto3/awscli, we should be fine on potential botocore conflicts. We can potentially use a package manager in the future to ensure we don't run into dependency conflicts

**Review by arjkesh — APPROVED** [review]

**Comment by arjkesh** [comment]
@jeet4320 - would it be possible to add sanity tests for boto3/awscli installation since we're adding new packages to the container?

can be simple, in the same vein as the emacs tests

```
@pytest.mark.model("N/A")
@pytest.mark.integration("awscli")
def test_awscli(mxnet_inference):
    """
    Ensure that awscli is installed on appropriate image
    """
    image = mxnet_inference
    ctx = Context()
    container_name = get_container_name("awscli", image)
    start_container(container_name, image, ctx)
    # Make sure the following emacs sanity tests exit with code 0
    run_cmd_on_container(container_name, ctx, "which aws")
    run_cmd_on_container(container_name, ctx, "aws --version")
```

```
@pytest.mark.model("N/A")
@pytest.mark.integration("boto3")
def test_awscli(mxnet_inference):
    """
    Ensure that boto3 is installed on appropriate image
    """
    image = mxnet_inference
    ctx = Context()
    container_name = get_container_name("awscli", image)
    start_container(container_name, image, ctx)
    # Make sure the following emacs sanity tests exit with code 0
    run_cmd_on_container(container_name, ctx, "python -c 'import boto3'")
```

File: test/dlc_tests/sanity/test_pre_release.py
Comment (arjkesh) [inline test/dlc_tests/sanity/test_pre_release.py:400]: nit: we may not make boto3 a requirement going forward, so let's call out that we are installing on MXNet inference containers in the docstring. It looks like we have boto3 in recent tensorflow2 inference images as well, so you could make this 

```
def test_boto3(inference):
"""
Ensure that boto3 is installed in relevant inference images
"""
fw, _ = get_framework_and_version_from_tag(inference)

# Note: boto3 is currently only regularly installed on tf and mxnet inference images
if fw not in ("tensorflow", "mxnet"):
    pytest.skip(f"Skipping test on unsupported framework {fw}")
```

File: test/dlc_tests/sanity/test_pre_release.py
Comment (arjkesh) [inline test/dlc_tests/sanity/test_pre_release.py:409]: probably actually better to do

```
run_cmd_on_container(container_name, ctx, 'import boto3', executable="python")
```

File: test/dlc_tests/sanity/test_pre_release.py
Comment (arjkesh) [inline test/dlc_tests/sanity/test_pre_release.py:382]: may want to skip this for huggingface

File: test/dlc_tests/sanity/test_pre_release.py
Comment (saimidu) [reply test/dlc_tests/sanity/test_pre_release.py]: @arjkesh @jeet4320 We need `boto3` and `botocore` to make the EC2 service calls in DLC telemetry scripts work.

Comment (arjkesh) [reply test/dlc_tests/sanity/test_pre_release.py]: > @arjkesh @jeet4320 We need `boto3` and `botocore` to make the EC2 service calls in DLC telemetry scripts work.

@saimidu @jeet4320 - I don't think this is accurate. We do not need boto3, just botocore for the script. Botocore is a downstream dependency of awscli.

Comment (jeet4320) [reply test/dlc_tests/sanity/test_pre_release.py]: awscli brings botocore, so we test awscli for all dlc images then

**Comment by saimidu** [comment]
LGTM. Let's merge this.

**Comment by arjkesh** [comment]
@jeet4320 can you make sure all appropriate builds run? the awscli test is run on every image so we should ensure no surprises before merge

**Review by saimidu — APPROVED** [review]
Please merge this only after the newly added tests succeed as per the CI job logs.

**Comment by jeet4320** [comment]
all sanity passed and mxnet sanity was also passed on earlier commit but was failing due to var missing. I fixed it. Merging it.

**Comment by jeet4320** [comment]
merging PR as mxnet sanity passed with oss complaince is failing which is intermittent and has passed on earlier commit

**Pull request merged**
